runtime/arithmetic/int.cpp

```diff
@@ -230,6 +231,7 @@ mpz_ptr hook_INT_log2(mpz_t a) {
 void extract(mpz_t result, mpz_t i, size_t off, size_t len) {
   size_t size = (len + LIMB_BITS - 1) / LIMB_BITS;
   mpz_init2(result, len+LIMB_BITS);
+  memset(result->_mp_d, 0, size);
```

Comment:
Why is this needed? mpz_init2 promises to set the number to zero. Is the problem from calling raw mpn functions later?

Reply:
Yeah pretty much. The issue is that if size > 0 and the final result of the operation is also zero, then the while loop at the end of this function will read uninitialized memory while trying to compute the appropriate value for _mp_size. properly speaking, we should probably be using mpz_limbs_write and mpz_limbs_finish, but this code also works fine (and I wrote it this way because I was aping the Zarith ocaml implementation)

Reply:
Hmm, this stuff *is* documented, though some of it is not guaranteed to be stable across versions. I suppose it's ok.